packages/good-design/src/apps/goodid/screens/OffersAgreement.tsx

```diff
@@ -1,19 +1,19 @@
-import React, { useContext } from "react";
+import React from "react";
 import { Center, Container, Text, VStack } from "native-base";
 import { useWizard } from "react-use-wizard";
+import { AsyncStorage } from "@gooddollar/web3sdk-v2";
 
-import { WizardContext } from "../../../utils/WizardContext";
 import { GoodButton, Image, Title } from "../../../core";
 
 import BillyGrin from "../../../assets/images/billy-grin.png";
 
 export const OffersAgreement = () => {
-  const { updateDataValue } = useContext(WizardContext);
   const { nextStep } = useWizard();
 
   // we need the offersAgreement acceptance for running checkAvailableOffers
-  const handleAccept = () => {
-    updateDataValue("segmentation", "offersAgreement", true);
+  const handleAccept = async () => {
+    // handle through controller
+    await AsyncStorage.setItem("goodid_permission", "true");
```

Comment:
what if user doesnt accept?

Reply:
Then we don't have the permission stored in localStorage.
seemed a bit redundant to explicitly store the value and set it to false

Either the permission exists or it doesn't.

Reply:
if you put false you know user actually went through this stage and actively selected not to share